CMakeLists.txt

```diff
@@ -4,21 +4,26 @@ project(nps_uw_sensors_gazebo)
 find_package(catkin REQUIRED tf)
 
 find_package(gazebo REQUIRED)
+find_package(roscpp REQUIRED)
+find_package(std_msgs REQUIRED)
 find_package(OpenCV)
+include_directories(${roscpp_INCLUDE_DIRS})
+include_directories(${std_msgs_INCLUDE_DIRS})
 include_directories(
   include
   ${catkin_INCLUDE_DIRS} 
   ${GAZEBO_INCLUDE_DIRS}
 )
 link_directories(${GAZEBO_LIBRARY_DIRS})
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GAZEBO_CXX_FLAGS}")
 
 catkin_package()
 
 ## Plugins
-add_library(nps_gazebo_ros_gpu_sonar_single_beam_plugin
-            src/nps_gazebo_ros_gpu_sonar_single_beam.cpp)
-target_link_libraries(nps_gazebo_ros_gpu_sonar_single_beam_plugin
-                      GpuRayPlugin ${catkin_LIBRARIES})
+add_library(gazebo_ros_pulse_lidar_plugin src/gazebo_ros_pulse_lidar.cpp)
+add_library(nps_gazebo_ros_gpu_sonar_single_beam_plugin src/nps_gazebo_ros_gpu_sonar_single_beam.cpp)
```

Comment:
Looks like src/nps_gazebo_ros_gpu_sonar_single_beam.cpp is not here, please check.

Reply:
Scroll down a few lines. We can keep these separate if you want. Does this compile for you?

Reply:
Oops, I reported the wrong filename.  Please check for presence of src/gazebo_ros_pulse_lidar.cpp

Reply:
ugh! Sorry, it should be there now.

Reply:
Fixed, good.